### ISSUE TITLE
[Feature Request] Method Customization

### ISSUE THREAD

**Issue by chamons** [Azure/autorest.csharp#1228]
Today there is no trivial way to say "I want this specific method as internal" outside a swagger transform to add x-accessibility for a specific language.

Maybe [CodeGenAccessibility] like [CodeGenSuppress]?

Originally from @KrzysztofCwalina

**Comment by KrzysztofCwalina** [comment]
> Maybe [CodeGenAccessibility] like [CodeGenSuppress]?

We should use C# code to control C# specific things. In this particular case, I would just allow to add an extern defintion to the project to change visibility, e.g. 

```csharp
extern internal virtual async Task<Response> GroupExistsAsync(string group, RequestOptions options = default);
```

**Comment by chamons** [comment]
Closed in favor of #1274
